### PR TITLE
Feature/tagging

### PR DESCRIPTION
**Pull request by jacobvillegas**
Delete works, add tag to an image is still not fully working

### CODE REVIEW

File: backend/flaskr/image_management/tagging_model.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+                conne.commit()
+                print(new_tag_ids)
+            print(f"Tag: {tag_id} added to photo: {img_id}")
+        except Exception as e:
+            print(e)
+            print(f'Addition of tag: failed')
+
+        finally:
+            cursor.close
```
Comment (jigneshsatam) [inline backend/flaskr/image_management/tagging_model.py:102]: close needs to be close()

File: backend/flaskr/image_management/tagging_model.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        new_tag_names =[]
+        new_tag_ids = []
+        tag_with_ids = []
+        tagging_ids=[]
+
+
+
+        try:
+            conne = Connect().get_connection()
+            cursor = conne.cursor()
```
Comment (jigneshsatam) [inline backend/flaskr/image_management/tagging_model.py:33]: Please move these code outside the try

File: backend/flaskr/image_management/tagging_model.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    @classmethod
+    def get_taged_img(cs, tag_id, img_id):
+
+        tagged_query = f'SELECT tag_id, img_id FROM tagging WHERE tag_id = {tag_id} AND img_id = {img_id};'
+
+        try:
+            conn = Connect().get_connection()
+
+            # Create cursor to perform database operations
+            cursor = conn.cursor()
```
Comment (jigneshsatam) [inline backend/flaskr/image_management/tagging_model.py:116]: same here

File: backend/flaskr/image_management/tagging_model.py [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+            cursor.execute(tagged_query)
+            #query_df = pd.DataFrame(cursor.fetchall())
+            tagging = []
+            #cursor.execute(query)
+            for row in cursor.fetchall():
+                tagging.append({"tag_id" : row[0],
+                                  "img_id": row[1]
+                                  })
+            cursor.close()
```
Comment (jigneshsatam) [inline backend/flaskr/image_management/tagging_model.py:128]: remove this `cursor.close()`

File: backend/flaskr/image_management/tagging_model.py [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+            return tagging
```
Comment (jigneshsatam) [inline backend/flaskr/image_management/tagging_model.py:129]: Add the return after finally

File: backend/flaskr/image_management/tagging_model.py [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+
+        # tag_names = ["'tag_new'", "'tag_new_2'"]
+        tag_names = []
+
+        tagging_ids= []
+
+        try:
+            # Create cursor to perform database operations
+            conn = Connect().get_connection()
+            cursor = conn.cursor()
```
Comment (jigneshsatam) [inline backend/flaskr/image_management/tagging_model.py:162]: same here

File: backend/flaskr/image_management/tagging_model.py [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+                conn.commit()
+                print(new_tag_ids)
+
+
+        except Exception as e:
+            print("Error::", e)
+            print(f"no image:{dir_id} with tag:{tags} ")
+
+        finally:
+            cursor.close
```
Comment (jigneshsatam) [inline backend/flaskr/image_management/tagging_model.py:257]: Parenthesis

File: backend/flaskr/image_management/tagging_model.py [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+    @classmethod
+    def delete_tags(cs, tag_id, img_id):
+       del_query = f'DELETE FROM tagging WHERE tag_id= {tag_id} AND img_id = {img_id};'
+
+
+       try:
+           success = False
+
+           conn = Connect().get_connection()
+           cursor = conn.cursor()
```
Comment (jigneshsatam) [inline backend/flaskr/image_management/tagging_model.py:271]: same here

**Review by jigneshsatam — COMMENTED** [review]
Please address the comments

File: backend/flaskr/image_management/tagging_model.py (hunk 1 above)
Comment (jacobvillegas) [reply backend/flaskr/image_management/tagging_model.py:102]: Thanks done!

File: backend/flaskr/image_management/tagging_model.py (hunk 2 above)
Comment (jacobvillegas) [reply backend/flaskr/image_management/tagging_model.py:33]: Done, thank you.

File: backend/flaskr/image_management/tagging_model.py (hunk 3 above)
Comment (jacobvillegas) [reply backend/flaskr/image_management/tagging_model.py:116]: Takan care of

File: backend/flaskr/image_management/tagging_model.py (hunk 4 above)
Comment (jacobvillegas) [reply backend/flaskr/image_management/tagging_model.py:128]: Removed, thank you.

File: backend/flaskr/image_management/tagging_model.py (hunk 5 above)
Comment (jacobvillegas) [reply backend/flaskr/image_management/tagging_model.py:129]: Thanks, taken care of.

File: backend/flaskr/image_management/tagging_model.py (hunk 6 above)
Comment (jacobvillegas) [reply backend/flaskr/image_management/tagging_model.py:162]: Thanks, done.

File: backend/flaskr/image_management/tagging_model.py (hunk 7 above)
Comment (jacobvillegas) [reply backend/flaskr/image_management/tagging_model.py:257]: Thank you, done

File: backend/flaskr/image_management/tagging_model.py (hunk 8 above)
Comment (jacobvillegas) [reply backend/flaskr/image_management/tagging_model.py:271]: done, thanks.

**Comment by jacobvillegas** [comment]
Changes were added

**Pull request closed without merging**